lib/datadog/core/telemetry/event.rb

```diff
@@ -313,6 +313,28 @@ def type
             'distributions'
           end
         end
+
+        # Telemetry class for the 'message-batch' event
+        class MessageBatch
+          attr_reader :events
+
+          def type
+            'message-batch'
+          end
+
+          def initialize(events)
+            @events = events
+          end
+
+          def payload(seq_id)
+            @events.map do |event|
+              {
+                request_type: event.type,
+                payload: event.payload(seq_id),
```

Comment:
Is it OK that seq id is the same for all of the events?

Reply:
Great question, I need to look it up in the spec.

Reply:
The spec says nothing about it and also it does not mention that `seq_id` should be part of telemetry events payloads at all (it is supposed to be part of common headers)! I will investigate why our payload generation code even depends on seq_id...

Reply:
I found where this seq_id is defined in specs: https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/conf_key_value.md
```
This field should incremented each time a new configuration key-value pair is applied in order to track the active set of configurations. It can be a global value across all configurations keys or local to each key..
```

This seq_id is in fact different from the seq_id that is used for events:
https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/development.md#required-top-level-json-keys

```
Counter that should be auto incremented every time an API call is being made (our pipeline may not always work in a sequential order.This field will help us identify situations where some messages were being lost. seq_id should start at 1 so that the go backend can detect between non set fields.
```

I will see how to decouple them from each other in our implementation

Reply:
Solved this by introducing another sequence in Core::Telemetry::Event class for the configuration changes